crates/client-executor/src/lib.rs

```diff
@@ -21,6 +21,35 @@ pub struct ContractInput<C: SolCall> {
     pub calldata: C,
 }
 
+sol! {
+    /// Public values of a contract call.
+    ///
+    /// These outputs can easily be abi-encoded, for use on-chain.
+    struct ContractPublicValues {
+        address contractAddress;
+        address callerAddress;
+        bytes contractCallData;
+        bytes contractOutput;
+        bytes32 blockHash;
+    }
+}
+
+impl ContractPublicValues {
+    /// Construct a new [`ContractPublicValues`]
+    ///
+    /// By default, commit the contract input, the output, and the block hash to public values of
+    /// our proof. More can be committed if necessary.
```

Comment:
nit: `the` proof. For certain applications, committing more data may be necessary.